README.md

```diff
@@ -93,17 +114,144 @@ run the hook from outside Vagrant).
 
 ## Adding zulint to a codebase
 
-TODO.  Will roughly include `pip install zulint`, copying an example
-`lint` script, and adding your rules.
+TODO: Make a pypi release
+
+Add `zulint` to your codebase requirements file or just do:
+
+```
+pip install zulint
+```
+
+See [example-lint](./example-lint) file for a simple example of adding zulint to
+your codebase. For a more advanced usage example, you can look at
+[Zulip's linter](https://github.com/zulip/zulip/blob/master/tools/lint).
 
+**NOTE**
+
+* Remember to mark your lint file executable using:
+
+```bash
+> chmod +x <YOUR LINT FILE>
+```
+
+* Add `<YOUR LINTER VARIABLE NAME>.do_lint()` add the end of your lint file.
 
 ## Adding third-party linters
 
-TODO: Document the linter_config API.
+First import the `LinterConfig` and initialize it with default arguments.
+You can then use the `external_linter` method to register the linter.
+
+eg:
+
+```python
+import argparse
+from zulint.command import add_default_linter_arguments, LinterConfig
+parser = argparse.ArgumentParser()
+# Add custom parser arguments here.
+
+add_default_linter_arguments(parser)
+args = parser.parse_args()
+
+linter_config = LinterConfig(args)
+
+# Add your external litner:
+linter_config.external_linter('eslint', ['node', 'node_modules/.bin/eslint',
+                                          '--quiet', '--cache', '--ext', '.js,.ts'], ['js', 'ts'],
+                              fix_arg='--fix',
+                              description="Standard JavaScript style and formatting linter"
+                              "(config: .eslintrc).")
+
+linter_config.do_lint()
+```
+
+Please make sure external linter (here `eslint`) is accessible via bash or in the
+virtual env where this linter will run.
```

Comment:
I think we should assume that the user is starting from our example, and have this just document how to add an external linter (and not e.g. show the argparse code, but just mention we automatically generate the include/exclude arguments)?  Otherwise we're just repeating most of the example's code.

Also, can you add a section explaining what the various options to `external_linter` do?  I think it can just be a bulleted list with a line per option.